ui/tests/integration/components/secret-engine/configure-wif-test.js

```diff
@@ -0,0 +1,901 @@
+          });
+          this.server.post('/identity/oidc/config', () => {
+            return overrideResponse(400, { errors: ['bad request'] });
+          });
+          await fillInAwsConfig('withWif');
+          await click(GENERAL.saveButton);
+          await click(SES.wif.issuerWarningSave);
+
+          assert.true(
+            this.flashDangerSpy.calledWith('Issuer was not saved: bad request'),
+            'Flash message shows that issuer was not saved'
+          );
+          assert.true(
+            this.flashSuccessSpy.calledWith(`Successfully saved ${this.id}'s configuration.`),
+            'Flash message shows that root was saved even if issuer was not'
+          );
+          assert.ok(
+            this.transitionStub.calledWith('vault.cluster.secrets.backend.configuration', this.id),
+            'Transitioned to the configuration index route.'
+          );
+        });
+
+        test('it surfaces the API error if config save fails, and prevents the user from transitioning', async function (assert) {
+          this.id = `aws-${this.uid}`;
+          this.displayName = 'AWS';
+          this.issuerConfig = createConfig(this.store, this.id, 'issuer');
+          this.config = this.store.createRecord('aws/root-config');
+          this.secondConfig = this.store.createRecord('aws/lease-config');
+          this.config.backend = this.secondConfig.backend = this.id;
+
+          await render(hbs`
+                <SecretEngine::ConfigureWif @backendPath={{this.id}} @displayName={{this.displayName}} @type='aws' @model={{this.config}} @issuerConfig={{this.issuerConfig}} @secondModel={{this.secondConfig}}/>
+              `);
+          this.server.post(configUrl('aws', this.id), () => {
+            return overrideResponse(400, { errors: ['bad request'] });
+          });
+          this.server.post(configUrl('aws-lease', this.id), () => {
+            assert.true(
+              true,
+              'post request was made to config/lease when config/root failed. test should pass.'
+            );
+          });
+          // fill in both lease and root endpoints to ensure that both payloads are attempted to be sent
+          await fillInAwsConfig('withAccess');
+          await fillInAwsConfig('withLease');
+          await click(GENERAL.saveButton);
+          assert.dom(GENERAL.messageError).exists('API error surfaced to user');
+          assert.dom(GENERAL.inlineError).exists('User shown inline error message');
+        });
+      });
+
+      module('Azure specific', function () {
+        test('it clears access type inputs after toggling accessType', async function (assert) {
+          this.id = `azure-${this.uid}`;
+          this.displayName = 'Azure';
+          this.issuerConfig = createConfig(this.store, this.id, 'issuer');
+          this.config = this.store.createRecord('azure/config');
+          this.config.backend = this.id;
```

Comment:
can these be set in the beforeEach for this module instead of configured in each test?